### PR TITLE
Feat(eos_designs)!: fabric variable for bgp default ipv4 unicast

### PR DESCRIPTION
**Pull request by carlbuchmann**
## Change Summary

Add fabric variable for `bgp default ipv4 unicast`, and enforce best practice (disable)

- updated documentation to remove references in bgp_defaults:

## Related Issue(s)

fixes aristanetworks/avd-internal#100

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

```yaml
bgp_default_ipv4_unicast: <bool> -> default false
```

## How to test

See molecule test results 

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by ClausHolbechArista — COMMENTED** [review]
Please add release-notes and porting guide.

File: ansible_collections/arista/avd/docs/porting-guides/4.x.x.md [hunk 1]
```diff
@@ -15,7 +15,29 @@ built in to `eos_designs`.
 
 ### Fabric variables
 
-**BGP peer groups variables:**
+**BGP variables:**
+
+A new key, `bgp_default_ipv4_unicast: <bool> -> default false` was introduced to implement the best practice of disabling the default activation of IPv4 unicast address-family on all IPv4 neighbors.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/porting-guides/4.x.x.md]:
```suggestion
A new key, `bgp_default_ipv4_unicast: <bool> -> default false` was introduced to implement the best practice of disabling the default activation of IPv4 unicast address-family on all IPv4 neighbors.
```
```suggestion
A new key, `bgp_default_ipv4_unicast: <bool> -> default false` was introduced to implement the best practice of disabling the default activation of IPv4 unicast address-family.
```

File: ansible_collections/arista/avd/docs/release-notes/4.x.x.md [hunk 2]
```diff
@@ -98,6 +98,12 @@ See details in the [Porting guide for AVD 4.x.x](../porting-guides/4.x.x.md#fabr
 
 See details in the [Porting guide for AVD 4.x.x](../porting-guides/4.x.x.md#fabric-variables).
 
+### change in defaults eos_designs in BGP variables
+
+A new key, `bgp_default_ipv4_unicast: <bool> -> default false` was introduced to implement the best practice of disabling the default activation of IPv4 unicast address-family on all IPv4 neighbors.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/release-notes/4.x.x.md]:
```suggestion
A new key, `bgp_default_ipv4_unicast: <bool> -> default false` was introduced to implement the best practice of disabling the default activation of IPv4 unicast address-family.
```

File: ansible_collections/arista/avd/docs/porting-guides/4.x.x.md [hunk 3]
```diff
@@ -15,7 +15,29 @@ built in to `eos_designs`.
 
 ### Fabric variables
 
-**BGP peer groups variables:**
+**BGP variables:**
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/porting-guides/4.x.x.md]:
```suggestion
#### BGP variables
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/base/avdstructuredconfig.py [hunk 4]
```diff
@@ -41,6 +43,11 @@ def router_bgp(self) -> dict | None:
             "as": self.shared_utils.bgp_as,
             "router_id": self.shared_utils.router_id,
             "bgp_defaults": bgp_defaults,
+            "bgp": {
+                "default": {
+                    "ipv4_unicast": bgp_default_ipv4_unicast,
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/base/avdstructuredconfig.py:48]: and if. you do this you can remove line 34/35
```suggestion
                    "ipv4_unicast": get(self._hostvars, "bgp_default_ipv4_unicast", default=False),
```

**Review by gmuloc — APPROVED** [review]

**Pull request merged**